syntax = "proto3";

package avatar_sync;

import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";

service AvatarSyncService {
  rpc SyncAvatarStream(stream AvatarSyncStreamRequest) returns (google.protobuf.Empty);
}

message AvatarSyncStreamRequest {
  oneof request_data {
    SyncConfig config = 1;        // 스트림 시작 시 설정 정보
    bytes audio_chunk = 2;        // TTS로부터 받은 오디오 청크
    VisemeData viseme_data = 3;   // (선택적) TTS에서 생성된 입술 모양 정보 (Viseme)
  }
}

message SyncConfig {
  string session_id = 1;
}

// 입술 모양(Viseme) 정보를 표현하는 메시지 예시
message VisemeData {
  string viseme_id = 1;
  google.protobuf.Timestamp start_time = 2;
  float duration_sec = 3;
}
